pkg/api/http_server.go

```diff
@@ -158,7 +160,7 @@ func ProvideHTTPServer(opts ServerOptions, cfg *setting.Cfg, routeRegister routi
 	dataSourcesService datasources.DataSourceService, secretsService secrets.Service, queryDataService *query.Service,
 	ldapGroups ldap.Groups, teamGuardian teamguardian.TeamGuardian, serviceaccountsService serviceaccounts.Service,
 	authInfoService login.AuthInfoService, resourcePermissionServices *resourceservices.ResourceServices,
-	notificationService *notifications.NotificationService, datasourcePermissionsService DatasourcePermissionsService) (*HTTPServer, error) {
+	notificationService *notifications.NotificationService, datasourcePermissionsService DatasourcePermissionsService, dashboardsnapshotsService dashboardsnapshots.Service) (*HTTPServer, error) {
```

Comment:
Think you need to use pointer receiver here since `dashboardsnapshots.Service` is a struct and not an interface.